internal/manager/pvcleanup/suite_test.go

```diff
@@ -0,0 +1,582 @@
+					},
+				},
+				Spec: corev1.PersistentVolumeSpec{
+					Capacity: corev1.ResourceList{
+						corev1.ResourceStorage: resource.MustParse("10Gi"),
+					},
+					PersistentVolumeReclaimPolicy: corev1.PersistentVolumeReclaimRetain,
+					AccessModes: []corev1.PersistentVolumeAccessMode{
+						corev1.ReadWriteOnce,
+					},
+					PersistentVolumeSource: corev1.PersistentVolumeSource{
+						CSI: &corev1.CSIPersistentVolumeSource{
+							Driver:       lvm.DriverName,
+							VolumeHandle: "test-volume-3",
+						},
+					},
+					NodeAffinity: &corev1.VolumeNodeAffinity{
+						Required: &corev1.NodeSelector{
+							NodeSelectorTerms: []corev1.NodeSelectorTerm{
+								{
+									MatchExpressions: []corev1.NodeSelectorRequirement{
+										{
+											Key:      lvm.TopologyKey,
+											Operator: corev1.NodeSelectorOpIn,
+											Values:   []string{"non-existent-node"},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+
+			Expect(k8sClient.Create(ctx, pv)).To(Succeed())
+
+			// Update status to Released to trigger reconciliation check
+			pv.Status.Phase = corev1.VolumeReleased
+			Expect(k8sClient.Status().Update(ctx, pv)).To(Succeed())
+
+			// Wait a bit to ensure controller has time to process
+			time.Sleep(time.Second * 2)
+
+			// Verify finalizers are still present (not removed)
+			var updatedPV corev1.PersistentVolume
+			Expect(k8sClient.Get(ctx, types.NamespacedName{Name: pv.Name}, &updatedPV)).To(Succeed())
+			Expect(updatedPV.Finalizers).To(ContainElement(PVProtectionFinalizer))
+
+			// Clean up
+			_ = k8sClient.Delete(ctx, pv)
+		})
+
+		It("Should not process PV that is not Released", func() {
+			pv := &corev1.PersistentVolume{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-pv-bound",
+					Finalizers: []string{
+						PVProtectionFinalizer,
+					},
+				},
+				Spec: corev1.PersistentVolumeSpec{
+					Capacity: corev1.ResourceList{
+						corev1.ResourceStorage: resource.MustParse("10Gi"),
+					},
+					PersistentVolumeReclaimPolicy: corev1.PersistentVolumeReclaimDelete,
+					AccessModes: []corev1.PersistentVolumeAccessMode{
+						corev1.ReadWriteOnce,
+					},
+					PersistentVolumeSource: corev1.PersistentVolumeSource{
+						CSI: &corev1.CSIPersistentVolumeSource{
+							Driver:       lvm.DriverName,
+							VolumeHandle: "test-volume-4",
+						},
+					},
+					NodeAffinity: &corev1.VolumeNodeAffinity{
+						Required: &corev1.NodeSelector{
+							NodeSelectorTerms: []corev1.NodeSelectorTerm{
+								{
+									MatchExpressions: []corev1.NodeSelectorRequirement{
+										{
+											Key:      lvm.TopologyKey,
+											Operator: corev1.NodeSelectorOpIn,
+											Values:   []string{"non-existent-node"},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+
+			Expect(k8sClient.Create(ctx, pv)).To(Succeed())
+
+			// Wait a bit to ensure controller has time to process
+			time.Sleep(time.Second * 2)
```

Comment:
Same comment as above:

We can use Eventually to make sure this won't ever have race condition in the test